client/web/src/components/views/device-details-view.tsx

```diff
@@ -38,11 +38,13 @@ export default function DeviceDetailsView({
             {!readonly && (
               <Button
                 sizeVariant="small"
-                onClick={() =>
+                onClick={() => {
+                  incrementMetric("web_client_deauth")
```

Comment:
Ah had meant to leave a comment here. Since we don't gracefully shut down currently / the web client itself is shutdown after hitting logout, we never seem to hit the `then` here. Had put the metric logging before the API call to get around this.

It might make sense to add an `/api/logout` handler on the go side that calls `logout` via the local client and add the metric logging there instead if we can have that done after we have successfully logged out but before things start shutting down.

Definitely like `logout` more than `deauth`. Based on your comment [here](https://github.com/tailscale/tailscale/pull/10462#discussion_r1416454685) maybe `web_client_node_disconnect` would disambiguate things even more?

Reply:
Ohh yes I see what you're saying in terms of it being unreachable. Sending the metric first sounds good to me then.

I like `node_disconnect`!